src/doc/rustc/src/platform-support/wasm32-unknown-emscripten.md

```diff
@@ -0,0 +1,157 @@
+# `wasm32-unknown-emscipten`
+
+**Tier: 2**
+
+The `wasm32-unknown-emscripten` target is a WebAssembly compilation target which
+uses the [Emscripten](https://emscripten.org/) compiler toolchain. Emscripten is
+a C/C++ toolchain designed to make it as easy as possible to port C/C++ code
+written for Linux to run on the web. It thus provides POSIX-compatible (musl) `libc`
+and `libstd` implementations and many Linux APIs, access to the OpenGL and SDL APIs,
+and the ability to run arbitrary JavaScript code, all based on web APIs using JS
+glue code. With the `wasm32-unknown-emscripten` target, Rust code can interoperate
+with Emscripten's ecosystem, C/C++ and JS code, and web APIs.
+
+One existing user of this target is the
+[`pyodide` project](https://pyodide.org/) which provides a Python runtime in
+WebAssembly using Emscripten and compiles Python extension modules written in Rust
+to the `wasm32-unknown-emscripten` target.
+
+If you want to generate a standalone bare-bones WebAssembly binary that does not
+require JS glue code and access to the web APIs and can thus run outside the web,
+the [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md) target may be better
+suited for you. However, [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md)
+only has partial support for the Rust standard library and does not (easily)
+support interop with C/C++ code.
```

Comment:
```suggestion
If you want to generate a standalone WebAssembly binary that does not
require access to the web APIs or the Rust standard library,
the [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md) target may be better
suited for you. However, [`wasm32-unknown-unknown`](./wasm32-unknown-unknown.md) does not (easily)
support interop with C/C++ code.
```
As an aside, someday someone should really add an option to make `wasm32-unknown-unknown` emit the clang ABI instead of the wasm-bindgen ABI to fix the interop problem...